### DISCUSSION TITLE
pyscript.web: Use something other than "innerHTML += text" to add text to an element.

### DISCUSSION THREAD

**Discussion by Dharengo** [pyscript/pyscript#2150] [Proposals]
If you do this, any elements that are already there will be completely replaced with fresh ones. This means that properties that are local to those elements, such as event listeners, are lost. Currently the only way I can think of to add plain text through the constructor or append() without replacing elements is to wrap it in a span tag.

I suggest document.createTextNode. In fact, having a python class to wrap TextNodes would be cool.

**Comment by WebReflection** [comment]
I believe the right tool for the job here is [element.insertAdjacentHTML](https://developer.mozilla.org/en-US/docs/Web/API/Element/insertAdjacentHTML) so that text/HTML can be appended after without trashing previous nodes.

**Reply by WebReflection** [reply]
@mchilvers if you want me to help here, happy to do so

**Reply by WebReflection** [reply]
https://github.com/pyscript/pyscript/pull/2151
